[WPE][GTK] Create GStreamerWebRTCProvider

philn opened the pull request:
#### 517e8e1a86e53b1b69b194353a42f683e92881d2
<pre>
[WPE][GTK] Create GStreamerWebRTCProvider
<a href="https://bugs.webkit.org/show_bug.cgi?id=243735">https://bugs.webkit.org/show_bug.cgi?id=243735</a>

Reviewed by Carlos Garcia Campos and Xabier Rodriguez-Calvar.

For the time being, create a GStreamerWebRTCProvider from the LibWebRTCProvider code unit. Ideally
this file should be renamed though.

* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCProvider.h:
(WebKit::createLibWebRTCProvider):

Canonical link: <a href="https://commits.webkit.org/253287@main">https://commits.webkit.org/253287@main</a>
</pre>

carlosgcampos commented on Source/WebKit/WebProcess/Network/webrtc/LibWebRTCProvider.h:
```diff
@@ -25,14 +25,21 @@
 
 #pragma once
 
+// FIXME: We should likely rename this header file to WebRTCProvider.h because depending on the
+// build configuration we create either a LibWebRTCProvider, or a GStreamerWebRTCProvider or
+// fallback to WebRTCProvider. This rename would open another can of worms though, leading to the
+// renaming of more LibWebRTC-prefixed files in WebKit.
```
I would open a bug for this.

calvaris replied:
I agree.

philn committed:
[WPE][GTK] Create GStreamerWebRTCProvider
https://bugs.webkit.org/show_bug.cgi?id=243735

Reviewed by Carlos Garcia Campos and Xabier Rodriguez-Calvar.

For the time being, create a GStreamerWebRTCProvider from the LibWebRTCProvider code unit. Ideally
this file should be renamed though.

* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCProvider.h:
(WebKit::createLibWebRTCProvider):

Canonical link: https://commits.webkit.org/253287@main
